docs/design/gcp-wif-support_design.md

```diff
@@ -0,0 +1,155 @@
+- Skip to [4.3.4.5. Installing the Data Protection Application
+](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html-single/backup_and_restore/index#oadp-installing-dpa_installing-oadp-gcp) to create Data Protection Application
+    
+    Note that the key for credentials should be `service_account.json` instead of `cloud` in the official documentation example.
+    ```yaml
+    apiVersion: oadp.openshift.io/v1alpha1
+    kind: DataProtectionApplication
+    metadata:
+      name: <dpa_sample>
+      namespace: openshift-adp
+    spec:
+      configuration:
+        velero:
+          defaultPlugins:
+          - openshift
+          - gcp
+      backupLocations:
+        - velero:
+            provider: gcp
+            default: true
+            credential:
+              key: service_account.json
+              name: cloud-credentials-gcp
+            objectStorage:
+              bucket: <bucket_name>
+              prefix: <prefix>
+      # Temporary image override while https://github.com/vmware-tanzu/velero-plugin-for-gcp/pull/142 not cherry-picked to Openshift
+      unsupportedOverrides:
+        gcpPluginImageFqin: ghcr.io/kaovilai/velero-plugin-for-gcp:file-wif
+    ```
+
+## Alternatives Considered
+- Using Google Config Connector on OpenShift to manage short-lived tokens.
+    - This would require another long lived token to be created and managed by the administrator, increasing the attack surface.
+    - There are pull requests put up during investigation of this alternative.
+        - https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/797
+        - https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/801
+
+## Security Considerations
+This proposal allows OADP Operator to depend on short lived credentials generated by the Cloud Credentials Operator. This is a more secure way to authenticate to GCP than using a long lived service account key.
+
+## Compatibility
+<!-- A discussion of any compatibility issues that need to be considered -->
+
+## Implementation
+<!-- A description of the implementation, timelines, and any resources that have agreed to contribute. -->
+
+velero-plugin-for-gcp update to support (stop panicking on) external_account (WIF) credentials https://github.com/vmware-tanzu/velero-plugin-for-gcp/pull/142
+
+OADP Operator will be updated to bind openshift service account token when WIF credentials is used. The following is done today for AWS STS credentials and will be extended to GCP WIF credentials.
+```go
+		veleroContainer.VolumeMounts = append(veleroContainer.VolumeMounts,
+			corev1.VolumeMount{
+				Name:      "bound-sa-token",
+				MountPath: "/var/run/secrets/openshift/serviceaccount",
+				ReadOnly:  true,
+			}),
+```
+
+<!-- ## Open Issues
+A discussion of issues relating to this proposal for which the author does not know the solution. This section may be omitted if there are none. -->
```

Comment:
Maybe add the standardized OLM workflow for GCP WIF issue here ?

Reply:
Cool. Thanks!